File: Assignment_12/frontend/src/services/ProductService.js
import http from "../http-common";

const getAll = () => {
    return http.get("/products");
};

const get = id => {
    return http.get(`/products/${id}`);
};

const create = data => {
    return http.post("/products", data);
};

const update = (id, data) => {
    return http.put(`/products/${id}`, data);
};

const remove = id => {
    debugger;
    return http.delete(`/products/${id}`);
};


const ProductService = {
    getAll,
    get,
    create,
    update,
    remove
};

export default ProductService;

File: Assignment_12/backend/src/main/java/san/edu/lab12/service/ProductService.java
package san.edu.lab12.service;

import san.edu.lab12.domain.Product;

import java.util.List;
import java.util.Optional;

public interface ProductService {
    public Product create(Product department);
    public Product update(Product department);
    public Product get(Long id);
    public Boolean delete(Long id);
    public List<Product> getAll();
}
